### ISSUE TITLE
Clipboard cleared when closing source window(Linux / Ubuntu 20.04-Gnome)

### ISSUE THREAD

**Issue by schmema** [hluk/CopyQ#1413]
**Describe the bug**
When closing the window where a text was copied from (Ctrl + c) the clipboard gets cleared.  This only happens when CopyQ is active. When closing CopyQ the clipboard works persistently.

**To Reproduce**
1. Copy something
2. Look in the history -> Entry is there
2. Close the window
3. Look in the history -> Entry is gone

**Expected behavior**
The clipboard entries should be stored persistently. 

**Screenshots**
![pic1](https://user-images.githubusercontent.com/8369447/85124559-628d0100-b22a-11ea-8ebc-993d1ac80ba9.jpg)
![pic2](https://user-images.githubusercontent.com/8369447/85124578-6ae53c00-b22a-11ea-9d57-8e47765e6a6f.jpg)

**Version, OS and Environment**
CopyQ Clipboard Manager v3.10.0
Qt: 5.12.5
Compiler: GCC
Arch: x86_64-little_endian-lp64
OS: Ubuntu 20.04 LTS

**Comment by hluk** [comment]
AFAIK, this is expected behavior. Clipboard on Linux/X11 is owned and provided by applications.

**Comment by kohnish** [comment]
As a workaround you can keep running application called xclipboard.

**Comment by rpaciorek** [comment]
using two clipboard managers seems a bit weird ... and maintaining the clipboard content in such cases is one of the basic functions of a clipboard manager ... so IMHO CopyQ should do it ... and this is possible with CopyQ command:

```
function overrideFunction(fn) {
    const oldFn = global[fn];
    global[fn] = function() {
        var selection = str(data(mimeClipboardMode)) == "selection"
        if (!hasData() && data(mimeText).length == 0) {
            if (selection) {
                tab('_s');
                copySelection(getItem(0));
            } else {
                tab('_c');
                copy(getItem(0));
            }
        } else {
            if (selection) {tab('_s'); } else {tab('_c');}
            var item = {}
            dataFormats().forEach(t => {item[t] = data(t)});
            if (size() > 0) {remove(0)}
            add(item)
            tab(config('clipboard_tab'));
            oldFn();
        }
    }
}

overrideFunction('onClipboardChanged');
```

it used two extra tabs to keep current content of selection and clipboard, because I don't see a way to distinguish them in the main (common) tab

Edit (2025-12-19): fixed issue with copying space characters (`hasData()` return false for a string containing only spaces)

**Comment by kohnish** [comment]
True, parcellite has an option called "Restore Empty" to enable the feature. I think copyq is the only clipboard manager that doesn't support it out of the box.

**Comment by Chealer** [comment]
Thank you very much for reporting @schmema 
This certainly does not occur on Windows, with CopyQ 13. Does this persist in a recent version? If so, could you retitle to something like:
> OS clipboard lost/cleared when closing its source window on Linux (X11, GNOME)

Having used GNU/Linux as [one of] my [main] desktop(s) for more than a decade, I am quite shocked to read that this would be standard X11 behavior. Even [Artem S. Tashkinov's _Major Linux Problems on the Desktop, 2023 edition_](https://itvision.altervista.org/why.linux.is.not.ready.for.the.desktop.current.html) does not make any mention of that among its numerous X11 issues. If I don't remember that, I suppose that's because even 2003 Klipper worked around that!🤔

@schmema: Please clarify what is lost; is it CopyQ's top entry, the OS clipboard, or both? Also, if it includes the OS clipboard, are you positive that "This only happens when CopyQ is active."?

<details>
  <summary>License</summary>
This comment and any other from myself on this ticket is offered under the terms of CC0 1.0.
</details>

**Comment by schmema** [comment]
Morning guys, thanks for looking into this. Since I opened this issue over five years ago, my workflow has changed significantly: I no longer use an X11-based OS or CopyQ in my daily life.

To be honest, this is no longer relevant to my current setup. I’ll leave it to your discretion whether you’d like to keep this open for others or close it out.
